runtime/jcl/common/getstacktrace.c

```diff
@@ -48,22 +48,42 @@ getStackTraceForThread(J9VMThread *currentThread, J9VMThread *targetThread, UDAT
 	j9object_t throwable = NULL;
 	J9StackWalkState walkState = {0};
 	UDATA rc = J9_STACKWALK_RC_NONE;
+
 #if JAVA_SPEC_VERSION >= 19
-	J9VMContinuation *continuation = targetThread->currentContinuation;
+	BOOLEAN isVirtual = IS_JAVA_LANG_VIRTUALTHREAD(currentThread, threadObject);
+	if (isVirtual) {
+		/* Do not spin when acquiring access, if acquire failed, return NULL.
+		 * The caller of getStackTraceImpl will handle if should retry or get stack using unmounted path.
+		 */
+		j9object_t carrierThread = (j9object_t)J9VMJAVALANGVIRTUALTHREAD_CARRIERTHREAD(currentThread, threadObject);
+		/* Ensure virtual thread is mounted and not during transition. */
+		if (NULL == carrierThread) {
+			goto done;
+		}
+		/* Gets targetThread from the carrierThread object. */
+		targetThread = J9VMJAVALANGTHREAD_THREADREF(currentThread, carrierThread);
+		Assert_JCL_notNull(targetThread);
+	}
 #endif /* JAVA_SPEC_VERSION >= 19 */
 	/* Halt the target thread */
 	vmfns->haltThreadForInspection(currentThread, targetThread);
```

Comment:
Added PUSH/POP for threadObject around the halt call. will look to merge `getStackTraceForThread` entirely into `Java_java_lang_Thread_getStackTraceImpl` in future refactor since it is the only caller. That will allow the code design to be simpler and require less conversion.